skrf/io/touchstone.py

```diff
@@ -471,3 +474,27 @@ def hfss_touchstone_2_network(filename, f_unit='ghz'):
     f,gamm,z0 = hfss_touchstone_2_gamma_z0(filename=filename)
     my_network.z0 = z0
     return(my_network)
+
+
+def read_zipped_touchstones(ziparchive, dir=""):
+    """
+    similar to skrf.io.read_all_networks, which works for directories but only for touchstones in ziparchives
+
+    Parameters
+    ----------
+    ziparchive : zipfile.ZipFile
+        an zip archive file, containing touchstone files and open for reading
+    dir : str
+        the directory of the ziparchive to read networks from, default is "" which reads only the root directory
+
+    Returns
+    -------
+    dict
+    """
+    networks = dict()
+    for fname in ziparchive.namelist():  # type: str
+        directory, filename = os.path.split(fname)
+        if dir == directory and fname[-4:].lower() in (".s1p", ".s2p", ".s3p", ".s4p"):
+            network = Network.zipped_touchstone(fname, ziparchive)
+            networks[network.name] = network
+    return networks
```

Comment:
yes i agree. i think we should also remove requirement that NetworkSet be homogenous (same #ports, frequency, etc). this might be moved to a 'check' that happens before statistics are computed.